subsys/net/lib/utils/addr_utils.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <net/socket.h>
+
+/* These utility functions are intended to be on top of POSIX API, and don't
+ * make sense if it's not available.
+ */
+#ifdef CONFIG_NET_SOCKETS_POSIX_NAMES
+
+const char *net_addr_str_find_port(const char *addr_str)
```

Comment:
> Why do we expose this function to user application, does it really need it? Could we just make it static?

Yes, the idea was to expose both functions to user apps, to give them the flexibility. For example, if a user apps needs address only in numeric format, it can use net_addr_str_find_port() to parse it into addr vs port substrings, and then use inet_pton(), and avoid dependency on getaddrinfo().

Reply:
I think we should keep this as static, it can be made global if really needed by someone. We should try to minimize the interfaces that we expose and this function is just too low level that application programmer would even find it in the API listing.

Reply:
> We should try to minimize the interfaces that we expose and this function is just too low level that application programmer would even find it in the API listing.

We should (finally) provide good and stable API set. And good API design exposes "atomic" functions, which can be combined by user to achieve any desired effect, while also providing higher-level convenience functions for commonly desired effect. That's exactly how this API is laid out. The API which offers only higher-level functionality and locks out user from using it, or require rewriting it completely to adjust some behavior, is well, bad API.

I don't see why all this concern about making that function static. Making it static is inviting someone to reinvent it in case of need. And the need can be easily anticipated.

Reply:
I just want to avoid exposing functions if there is no real need for them. It is a pain to decprecate them if we find out we need to tweak the API or remove them.

Reply:
> I just want to avoid exposing functions if there is no real need for them.

While this PR is titled as SNTP-related, one of the subtasks was to start laying out helper API for apps (including POSIX apps) to reuse, instead of coding it again and again. So I did spend mental effort on factoring out that func to be (re)usable by other parts. That's why I'm concerned that this effort goes to /dev/null by request to static'ize it.

> It is a pain to decprecate them if we find out we need to tweak the API or remove them.

So again, that func just does a useful thing on its own (parses address in format host[:host]), and intended for reuse, so I don't see it being under a reasonable risk of deprecation. (I'd be more concerned with visibly adhoc codes like https://github.com/zephyrproject-rtos/zephyr/pull/15636 - that doesn't implement general enough API, so would be subject of long incremental tweaks and partial deprecation, or will need to be replaced with a new more generic impl).


That said, if whole this discussion doesn't convince you, and you'd rather see "static" now, hopefully removed later, I'd rather move on with the matter.

Reply:
This is not that big of an issue. I just have been burned before because of introducing some functions that were "useful" too, but at the end they were really not useful after all.